### PR TITLE
Rename job.py to fm_step.py and Job to ForwardModelStep

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves eye itch

**Approach**
`nvim`

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/_ert/forward_model_runner/fm_step.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def killed_by_oom(pids: Sequence[int]) -> bool:
     return False
 
 
-class Job:
+class FMstep:
```
Comment (yngve-sk) [inline src/_ert/forward_model_runner/fm_step.py:79]: Maybe a nitpick, but I think the S should be big too?

Comment (berland) [reply src/_ert/forward_model_runner/fm_step.py:79]: It probably should. Potentially go all the way and write it as ForwardModelStep? That object name is actually not in use anywhere else in the codebase (although we do use it as a prefix for some other object names)

Comment (yngve-sk) [reply src/_ert/forward_model_runner/fm_step.py:79]: ForwardModelStep sounds better to me

**Review by eivindjahren — COMMENTED** [review]
:+1:

**Review by yngve-sk — APPROVED** [review]
LGTM. Final nitpick; would consider renaming the file fm_step.py to forward_model_step.py for consistency, but this is a good change regardless

**Pull request merged**